### PR TITLE
Update docs

### LINKED ISSUE #6326: Improve ModelAutocompleteFilter
## Feature Request

Using prefixes for resolve field name to searching entities from related admin.

Form here:
https://github.com/sonata-project/SonataAdminBundle/issues/6300#issuecomment-678812600

First question is: available prefix list it should be suggest in filter or is just hidden feature?

### PR DESCRIPTION
**Pull request by kirya-dev**
Update docs about `callback`:
https://github.com/kirya-dev/SonataAdminBundle/blob/update-autocomplete-docs/docs/reference/form_types.rst

Closes #6326

### CODE REVIEW

File: docs/reference/form_types.rst [hunk 1]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
-                    ->setParameter('barValue', $admin->getRequest()->get('bar'))
-                ;
-                $datagrid->setValue($property, null, $value);
-            },
-        ])
-    ;
+      $formMapper
+          ->add('category', ModelAutocompleteType::class, [
+              'property' => 'title',
+              'callback' => function ($admin, $property, $value) {
```
Comment (phansys) [inline docs/reference/form_types.rst:273]:
```suggestion
              'callback' => static function (AdminInterface $admin, string $property, $value): void {
```

File: docs/reference/form_types.rst [hunk 2]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
+      ;
+
+  If you want to dynamically change a ``property`` to something else, you can use prefix.
+  When you type **id: 20** property is id.
+  When you type: **username: awesome_user_name** property is username::
+
+      $formMapper
+          ->add('category', ModelAutocompleteType::class, [
+              'property' => 'title',
+              'callback' => function ($admin, $property, $value) {
```
Comment (phansys) [inline docs/reference/form_types.rst:292]:
```suggestion
              'callback' => static function (AdminInterface $admin, string $property, string $value): void {
```

File: docs/reference/form_types.rst [hunk 3]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
+                      ->andWhere($queryBuilder->getRootAlias() . '.foo=:barValue')
+                      ->setParameter('barValue', $admin->getRequest()->get('bar'))
+                  ;
+                  $datagrid->setValue($property, null, $value);
+              },
+          ])
+      ;
+
+  If you want to dynamically change a ``property`` to something else, you can use prefix.
+  When you type **id: 20** property is id.
```
Comment (phansys) [inline docs/reference/form_types.rst:286]:
```suggestion
  When you type **id: 20** property is "id".
```

File: docs/reference/form_types.rst [hunk 4]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
+  When you type: **username: awesome_user_name** property is username::
```
Comment (phansys) [inline docs/reference/form_types.rst:287]:
```suggestion
  When you type: **username: awesome_user_name** property is "username"::
```

**Comment by jordisala1991** [comment]
You need to rebase

**Review by VincentLanglet — APPROVED** [review]

File: docs/reference/form_types.rst [hunk 5]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
+                  $queryBuilder
+                      ->andWhere($queryBuilder->getRootAlias() . '.foo=:barValue')
+                      ->setParameter('barValue', $admin->getRequest()->get('bar'))
+                  ;
+                  $datagrid->setValue($property, null, $value);
+              },
+          ])
+      ;
+
+  If you want to dynamically change a ``property`` to something else, you can use prefix.
```
Comment (greg0ire) [inline docs/reference/form_types.rst:285]:
```suggestion
  If you want to dynamically change the ``property`` being filtered on to something else,
  you can use a prefix system, as follows.
```

File: docs/reference/form_types.rst [hunk 6]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
+  When you type **id: 20** property is "id".
```
Comment (greg0ire) [inline docs/reference/form_types.rst:286]:
```suggestion
  When the user types **id: 20** the property used for filtering is "id".
```

File: docs/reference/form_types.rst [hunk 7]
```diff
@@ -267,20 +267,40 @@ The available options are:
...
+  When you type: **username: awesome_user_name** property is "username"::
```
Comment (greg0ire) [inline docs/reference/form_types.rst:287]:
```suggestion
  When they type **username: awesome_user_name**, it will be "username"::
```

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @kirya-dev !

**Pull request merged**